airflow/providers/google/cloud/operators/sql_to_gcs.py

```diff
@@ -292,4 +292,4 @@ def _upload_to_gcs(self, files_to_upload):
             hook.upload(self.bucket, tmp_file.get('file_name'),
                         tmp_file.get('file_handle').name,
                         mime_type=tmp_file.get('file_mime_type'),
-                        gzip=self.gzip if tmp_file.get('file_name') == self.schema_filename else False)
+                        gzip=self.gzip if tmp_file.get('file_name') != self.schema_filename else False)
```

Comment:
Should we add a warning in init if both gzip and schema are provided?
```
:param gzip: Option to compress file for upload (does not apply to schemas).
```

Reply:
I don't see an issue there, you can define the schema and add the option to `gzip=True`. It'll perform what is documented `compress file for upload (does not apply to schemas)`, the fact that the schema is passed manually do not interfere, the schema file is created only if a schema filename is provided:

```python
        # If a schema is set, create a BQ schema JSON file.
        if self.schema_filename:
            self.log.info("Writing local schema file")
            files_to_upload.append(self._write_local_schema_file(cursor))
```

If `schema_filename` is set, then the file is created and the documentation still applies.